models/Track.js

```diff
@@ -46,6 +46,30 @@ module.exports = function(sequelize, DataTypes) {
   // INSTANCE
   //---------------
 
+  //add or replace a tag, such that this track only has 1 animal tag by this user
+  //and no duplicate tags
+  Track.prototype.replaceTag = async function(tag) {
+    const trackTags = await this.getTrackTags({
+      where: { UserId: tag.UserId, automatic: tag.automatic }
+    });
+    const existingTag = trackTags.find(function(uTag) {
+      return uTag.what == tag.what;
+    });
+    if (existingTag) {
+      return;
+    } else if (trackTags.length > 0 && !tag.isAdditionalTag()) {
+      const existingAnimalTags = trackTags.filter(function(uTag) {
+        return !uTag.isAdditionalTag();
+      });
+
+      for (let i = 0; i < existingAnimalTags.length; i++) {
+        await existingAnimalTags[i].destroy();
+      }
+    }
+
+    await tag.save();
```

Comment:
Can this be done inside a transaction to avoid problems with multiple replaceTag requests for a track coming in at around the same time?